taf/repository_tool.py

```diff
@@ -264,38 +277,41 @@ def add_targets(self, data, targets_role='targets', files_to_keep=None):
       files_to_keep = []
     # leave all files required by the framework and additional files specified by the user
     files_to_keep.extend(self._required_files)
-
+    # add all repositories defined in repositories.json to files_to_keep
+    files_to_keep.extend(self._get_target_repositories())
     # delete files if they no longer correspond to a target defined
     # in targets metadata and are not specified in files_to_keep
-    for root, _, files in os.walk(self.targets_path):
+    for root, _, files in os.walk(str(self.targets_path)):
       for filename in files:
-        filepath = os.path.join(root, filename)
-        if filepath not in data and filename not in files_to_keep:
-          os.remove(filepath)
+        filepath = Path(root) / filename
+        file_rel_path = str(Path(os.path.relpath(str(filepath), str(self.targets_path))).as_posix())
```

Comment:
So, based on previous comment, I think this could simplify things (since you introduced pathlib):

```python
for path in self.targets_path.rglob("*"):
    file_path = path.resolve()
    file_rel_path = str(path)
```

Could you try it out if you have tests?

Reply:
`rglob` seems to work, reworked this loop \